contrib/go-chi/chi/option.go

```diff
@@ -22,7 +22,10 @@ type config struct {
 type Option func(*config)
 
 func defaults(cfg *config) {
-	cfg.serviceName = "chi.router"
+	cfg.serviceName = globalconfig.ServiceName()
```

Comment:
Same here I guess. It's odd to me to assign something to a variable and then check if that variable is empty, when in fact we are actually checking the value of the assignment for empty. Does that even make sense?

Reply:
I can see it either way. If you *expect* the integration to get its name from `globalconfig.ServiceName()`, and consider `"chi.router"` a fallback, then this way makes sense. If you consider `"chi.router"` a default, which gets overridden by `globalconfig.ServiceName()`, then it makes sense to do it the way you suggest. Both have the same effect, it's just about how the reader interprets the intent. "Service name is the global service name. If it's empty, then use X" vs "Service name is X. If the user set a global name, then use that."

I will change it as you suggest, since I am not partial to one way or the other.